### PR TITLE
Customisable getRandom() function

### PR DESCRIPTION
**Pull request by allmywant**

### CODE REVIEW

File: tests/test_suite.html [hunk 1]
```diff
@@ -404,6 +404,71 @@
             w6CssInjected.should.equal(true);
         });
 
+        // decisionAdapter
+        var w7DecisionAdapterCalled = false;
+        Mojito.options.decisionAdapter = function(testObject)
+        {
+            w7DecisionAdapterCalled = true;
+            return 0.8;
+        };
```
Comment (kingo55) [inline tests/test_suite.html]: Excellent... this is awesome @allmywant

File: lib/mojito.js [hunk 2]
```diff
@@ -506,11 +525,19 @@ Mojito = (function ()
                 return val;
             },
             /**
-             * get random value from different engine
+             * get random value from decisionAdapter
              */
             getRandom: function()
             {
-                return this.getSeededRNGRandom();
+                if (!('decisionCnt' in this.options))
```
Comment (kingo55) [inline lib/mojito.js]: Sorry to change my mind here - could you please rename `decisionCnt` to `decisionIdx`? 

`Cnt` is a little too close to a bad word in English and the more I look at it, the more I want to distance our code from it.

File: lib/mojito.js [hunk 3]
```diff
@@ -129,6 +130,24 @@ Mojito = (function ()
                 }
             }
 
+            // decision adapter if TEST adapter defined, use it;
+            // else if GLOBAL adapter defined, use that;
+            // else use DEFAULT adapter 
+            if (!this.options.decisionAdapter)
```
Comment (kingo55) [inline lib/mojito.js:136]: Just to confirm your the logic above. Is this the test object's `decisionAdapter` or the global `decisionAdapter`?

As you mention, we should prioritise:

1. Test object `decisionAdapter`
2. Global `decisionAdapter`
3. Default to `seededRandom`

File: tests/test_suite.html [hunk 4]
```diff
@@ -404,6 +404,71 @@
...
+                }
+            }
+        });
+
+        it('Mojito.options.decisionAdapter was called', function () {
+            w7DecisionAdapterCalled.should.equal(true);
+        });
+
+        var w8ControlChosenByDecisionAdapter = false;
+        Mojito.options.decisionAdapter = function(testObject)
```
Comment (kingo55) [inline tests/test_suite.html]: For both tests, could you please add the `decisionAdapter` to the testObject directly?

I think there are some tests earlier on that activate after these do and therefore may be influenced from setting decisionAdapters here.

**Review by kingo55 — CHANGES_REQUESTED** [review]
Just two changes from me here - I'll have a proper test of it. @dapperdrop may have some other comments too:

1. Change `decisionCnt` to `decisionIdx`
2. Assign the `decisionAdapter` at the test object rather than globally

**Review by dapperdrop — APPROVED** [review]

File: lib/mojito.js (hunk 2 above)
Comment (allmywant) [reply lib/mojito.js]: @kingo55 
You're right, I will change it to ```decisionIdx```.

File: lib/mojito.js (hunk 3 above)
Comment (allmywant) [reply lib/mojito.js:136]: @kingo55 
Yes, you're right. The ```decisionAdapter``` will be assigned to the test object finally even we defined it with: Mojito.options.decisionAdapter = function() {}

Also, we can define different decisionAdapters for different tests, e.g:

```
Mojito.addTest({
    id: 'w51',
    name:'form-error',
    testScope:"user",
    sampleRate: 1,
    options: {
        debug: true, cookieDuration: 90,
        decisionAdapter: function (test)
        {
            if (test.options.decisionCnt === 0)
            {
                return 0.8;
            }
            return 0.6;
        }
    },
    state: 'live',
    recipes:{
        "0":{
            name: "Original",
        },
        "1":{
            name: "Variation A"
        }
    }
});
```

In this case, w51 test has its own ```decisionAdapter``` even we have the global decisionAdapter defined using Mojito.options.decisionAdapter = function() {}

Comment (kingo55) [reply lib/mojito.js:136]: Thanks...

File: lib/mojito.js (hunk 2 above)
Comment (allmywant) [reply lib/mojito.js]: Fixed.

File: tests/test_suite.html (hunk 4 above)
Comment (allmywant) [reply tests/test_suite.html]: Fixed.

**Review by kingo55 — APPROVED** [review]

**Comment by kingo55** [comment]
Thanks for the PR @allmywant !

**Pull request merged**